src/regfunc.jl

```diff
@@ -0,0 +1,349 @@
+#-----------------------------------------------------------
+#-----------------------------------------------------------
+# Define function to access windows registry
+# -- part of WinReg Module --
+#
+#File: regutils.jl
+#Author: Leonardo Cocco
+#Creation Date: 11-04-2016
+#Last update 21-11-2016
+#ver: 0.11
+#----------------------------------------------------------
+#----------------------------------------------------------
+
+
+"""Read a value with `valuename` into `key`. 
+`key` can be specified using abbreviated form (e.g. 'HKEY_LOCAL_MACHINE' -> 'HKLM').
+Return type in Julia is based onto value type stored in registry; returns `nothing` (type is `Void`) if key not found"""
+function regread(key::AbstractString, valuename::AbstractString)
+	local root = getroothkey(key)
+	if root == 0
+		return nothing
+	end
+	local subkey = getsubkey(key)
+	local hk::HKEY = 0
+	local len = Ref{DWORD}(0)
+	local typ = Ref{DWORD}(0)
+	
+	hk = regopenkeyex(root, subkey, REG_SAM.KEY_READ)
+	if hk != 0
+		#get buffer dim
+		local res = ccall((:RegQueryValueExW, advapi), stdcall,
+		  LONG,
+		  (HKEY, LPCWSTR, LPDWORD, LPDWORD, LPBYTE, LPDWORD),
+		  hk, transcode(Cwchar_t, valuename), C_NULL, typ, C_NULL, len )
+		if res == ERROR_SUCCESS
+			buf = Vector{UInt8}(len[])
+			len = Ref{DWORD}(length(buf))
+			
+			res = ccall((:RegQueryValueExW, advapi), stdcall,
+			  LONG,
+			  (HKEY, LPCWSTR, LPDWORD, LPDWORD, LPBYTE, LPDWORD),
+			  hk, transcode(Cwchar_t, valuename), C_NULL, typ, pointer(buf), len )
+			
+			regclosekey(hk)
+			
+			if res != ERROR_SUCCESS
+				return nothing
+			end
+		else
+			regclosekey(hk)
+			
+			return nothing
+		end
+	else
+		return nothing
+	end
+	
+	if typ[] == REG_TYPE.REG_SZ || typ[] == REG_TYPE.REG_EXPAND_SZ || typ[] == REG_TYPE.REG_LINK
+		local wbuf = reinterpret(Cwchar_t, buf)
+		return transcode(String, wbuf[1:div((len[]-2),2)])
```

Comment:
[According to the docs](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724911(v=vs.85).aspx), the result may or may not be null terminated.